### DISCUSSION TITLE
Allow setting refetchInterval on useContractRead

### DISCUSSION THREAD

**Discussion by brunowowk** [wevm/wagmi#2394] [Ideas]
In my use-case (mint page), I need somewhat fresh data (mint stats), but doesn't need to be always up-to-date with the latest block.
If I enable `watch`, it starts polling each 4s for the latest block, which burns through my alchemy quota pretty quickly.

Exposing the useQuery parameters https://github.com/wagmi-dev/wagmi/discussions/882 would be a good way to solve this, but I get that there are probably some edge cases to handle before doing that.

I suggest exposing just the `refetchInterval` parameter, as a way to give users full control over how fresh the data needs to be, since this is a major factor in node quota usage

**Comment by Drewsapple** [comment]
Is `staleTime` what you're looking for?

**Reply by brunowowk** [reply]
`staleTime`, by itself, would only refetch if the query is remounted. `refetchInterval` is required for polling:
https://tanstack.com/query/v4/docs/react/guides/important-defaults#:~:text=Stale%20queries%20are,a%20refetch%20interval

**Comment by jxom** [comment]
Could also set the global `pollingInterval` on `createConfig`, but agree, we could add a numeric option to `watch` so you can modify the polling interval per Hook.

**Reply by gpxl-dev** [reply]
Exposing `refetchInterval` would be more powerful than this due to this behaviour:

> If set to a function, the function will be executed with the latest data and query to compute a frequency

This is pretty useful - I can think of quite a few contract reads (e.g. `claimed` booleans) that once they reach a certain state you know you don't really want to ever worry about checking again.

For these we can use extremely high `cacheTime` and `staleTime` values, and then rely on the `refetchInterval` to fetch when it is not in its locked in state.

In general my view is to expose as many of the react-query standard query options as possible

**Comment by bjfresh** [comment]
Just came looking for this, as well. Would love to see it in the pipeline!

**Reply by brunowowk** [reply]
SPOILER ALERT

They're making wagmi fully react-query compatible on v2. Currently in alpha
https://github.com/wagmi-dev/wagmi/blob/9f3282cb39bfc40a5e7cd760043da74624b80c4d/docs/react/guides/migrate-from-v1-to-v2.md#moved-tanstack-query-to-peer-dependencies

**Reply by vantrung1408** [reply]
@volkyeth nice, i wondering when v2 will be released?

**Reply by brunowowk** [reply]
@vantrung1408 it's live, in alpha: https://github.com/wagmi-dev/wagmi/discussions/3068
Not sure when we should expect the beta
